mlflow/utils/_capture_modules.py

```diff
@@ -111,12 +111,14 @@ def main():
     # Mirror `sys.path` of the parent process
     sys.path = json.loads(args.sys_path)
 
-    if flavor == mlflow.spark.FLAVOR_NAME and is_in_databricks_runtime():
+    if flavor == mlflow.spark.FLAVOR_NAME:
         # Clear 'PYSPARK_GATEWAY_PORT' and 'PYSPARK_GATEWAY_SECRET' to enforce launching a new JVM
         # gateway before calling `mlflow.spark._load_pyfunc` that creates a new spark session
         # if it doesn't exist.
         os.environ.pop("PYSPARK_GATEWAY_PORT", None)
         os.environ.pop("PYSPARK_GATEWAY_SECRET", None)
+
```

Comment:
Could you add a logging here to log all environment variables,
and let user rerun and provide the logged environment variables, which helps debugging :)

Reply:
Do we really need to log all environment variables?

Reply:
Just for debugging ... we should remove it before merging

Reply:
added temporary logging with notes to remove